core/shared/src/main/scala/zio/prelude/ZValidation.scala

```diff
@@ -0,0 +1,1433 @@
+package zio.prelude
+
+import zio.prelude.ZValidation._
+import zio.test.Assertion
+import zio.{Chunk, IO, ZIO}
+
+import scala.util.Try
+
+/**
+ * `ZValidation` represents either a success of type `A` or a collection of one
+ * or more errors of type `E` along with in either case a log with entries of
+ * type `W`. Unlike `Either`, `ZValidation` does not "short circuit" on
+ * failures and instead allows accumulating multiple errors. This can be
+ * particularly useful in validating data, where we want to attempt to validate
+ * all of the data and retain information about all errors that arose, rather
+ * than failing at the first error.
+ */
+sealed trait ZValidation[+W, +E, +A] { self =>
```

Comment:
I could be worthwhile to put `log` on `ZValidation`, since it's supposed to be available for any case.
```suggestion
sealed abstract case class ZValidation[+W, +E, +A](log: Chunk[W]) { self =>
```

Reply:
I added a `getLog` operator since we already use `log` for appending to the log.

Reply:
Oh, OK then :+1: